### PR TITLE
Madhava/fix torch 1.6.0

### PR DESCRIPTION
**Pull request by madhavajay**
## Description
This PR lowers the torch requirements to torch==1.6.0.

## Affected Dependencies
None

## How has this been tested?
Tests were failing on python 3.6 and now they pass.
CI for all three python versions has been added.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/sympc/utils/utils.py [hunk 1]
```diff
@@ -100,9 +101,15 @@ def wrapper(
...
 
-        with Executor(
-            max_workers=nr_parties, initializer=initializer, initargs=(loop,)
-        ) as executor:
+        ex_kwargs: Dict[str, Any] = {}
+        ex_kwargs["max_workers"] = nr_parties
+
+        # disable additional args in python 3.6
+        if sys.version_info >= (3, 7):
+            ex_kwargs["initializer"] = initializer
```
Comment (gmuraru) [inline src/sympc/utils/utils.py:109]: Hmm, and if we have python3.6 I think the Executor would not work? (because it needs that main event loop?)

**Comment by madhavajay** [comment]
I think python 3.6 is possible: https://idolstarastronomer.com/two-futures.html

Code would look something like this but I don't have much asyncio experience so I will park this for the moment.
```
if sys.version_info >= (3, 7):
    ex_kwargs["initializer"] = initializer
    ex_kwargs["initargs"] = (loop,)

    with Executor(**ex_kwargs) as executor:
        for i in range(nr_parties):
            _args = args[i]
            _kwargs = kwargs.get(i, {})
            futures.append(executor.submit(funcs[i], *_args, **_kwargs))

else:
    with Executor(**ex_kwargs) as executor:
        for i in range(nr_parties):
            _args = args[i]
            _kwargs = kwargs.get(i, {})
            futures.append(
                await loop.run_in_executor(executor, funcs[i], *_args, *_kwargs)
            )
```

File: src/sympc/utils/utils.py (hunk 1 above)
Comment (madhavajay) [reply src/sympc/utils/utils.py]: You are right, i think its possible, see comment but I will leave it for the time being and re-address later if we have time.

**Review by LaRiffle — APPROVED** [review]

**Comment by gmuraru** [comment]
@madhavajay does this work? (it should be tested with the notebooks from the SMPC folder from Syft)

**Comment by madhavajay** [comment]
Yes, the latest 0.4 branch with this branch and torch 1.6.0 works on the `examples/secure-multi-party-computation/Duet/1-DS-1-DO` notebooks.
```
$ pip freeze | grep torch
torch==1.6.0
torchcsprng==0.1.2
torchvision==0.7.0
```
Screen Shot 2021-01-06 at 2 29 05 pm

**Pull request merged**